packages/skill-lesson/path-to-purchase/product-tiers.tsx

```diff
@@ -21,8 +21,28 @@ export const PricingTiers: React.FC<
   couponIdFromCoupon,
   allowPurchase,
 }) => {
-  const {redeemableCoupon, RedeemDialogForCoupon, validCoupon} =
-    useCoupon(couponFromCode)
+  const restrictedToProduct = couponFromCode?.restrictedToProductId
+    ? products.find(
+        (product) => product.productId === couponFromCode.restrictedToProductId,
+      )
+    : undefined
+
+  const productMetadata = restrictedToProduct
+    ? {
+        ...restrictedToProduct,
+        id: restrictedToProduct.productId,
+        image: {
+          ...restrictedToProduct.image,
+          width: 132,
+          height: 112,
+        },
+      }
+    : undefined
```

Comment:
This is how we look up the product metadata needed to pass in to `useCoupon` for the Redeem dialog to work. It looks up the product that matches the coupon restriction.